add header mdx (#310)
allenai/varnish-legacy: demo/src/pages/Header/page.mdx

---
type: Layout
title: AI2 Header
cols: 1
iconSrc: ./header-footer-14px.svg
---
import defaultDemo from './demo/default.mdx'; // crazy, we must import for the require to work
// these two lines grab the mdx files in the local demo dir and make examples for each
const mdxs = require.context('./demo', false, /\.(mdx)$/);
const examples = mdxs.keys().map(mdxs);

## AI2 Header

> This component renders a document header that is unified for all AI2 properties.

Reviewer: ```suggestion
This component renders a document header that is consistent across AI2 properties.
```

### When To Use

> You should put this at the top of the page. It must be present on every page of your application.

Reviewer: I think we may only require the `<AI2Banner />`. The `<Header />` is nice too, but I think there's cases where it doesn't make sense to use (like PAWLS).

Reviewer: Yea, it is -- but I think sometimes it makes sense for apps to derivate a bit. In this case we wanted to make as much vertical real-estate available for the PDF as possible, so having a large header at the top that takes up real-estate (with nothing but a logo in it), didn't make sense.

Author: updated

### Examples

<Examples cols={frontMatter.cols} examples={examples} />

### API

| Property      | Description                                   | Type    | Default |
| ------------- | --------------------------------------------- | ------- | ------- |

> | alwaysVisible | prevent top banner from hiding when scrolling | boolean | false   |

Reviewer: ```suggestion
| alwaysVisible | Prevents the AI2 banner from disappearing when the page is scrolled | boolean | false   |
```
